flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -457,7 +499,19 @@ protected int stop(String[] args) throws Exception {
 
 		final CustomCommandLine<?> activeCommandLine = getActiveCustomCommandLine(commandLine);
 
-		final ClusterClient client = activeCommandLine.retrieveCluster(commandLine, configuration, configurationDirectory);
+		final ClusterDescriptor<?> clusterDescriptor = activeCommandLine.createClusterDescriptor(
```

Comment:
It seems that there is a pattern of getting the `clusterDescriptor` and and the `client` for all the commands. Also shutting down the client and closing the descriptor is repeated. Imo we should extract the pattern to avoid code duplication.

Reply:
You're right. I would like to postpone addressing this issue until we've introduce the typed cluster id. Otherwise I fear that it would inflict quite some merge conflicts.

Reply:
ok